### ISSUE TITLE
Removing the configuration of a flag, enables it.

### ISSUE THREAD

**Issue by hamamou** [launchdarkly/terraform-provider-launchdarkly#198]
Hello, 
Let's say we have a feature flag setup this way:
```
resource "launchdarkly_feature_flag" "my-flag" {
    project_key    = local.project_key
    key            = "your-feature-flag-key"
    name           = "Your feature flag name"
    description    = "Your feature flag short description"
    variation_type = "boolean"
    tags           = ["some", "tags"]
    temporary      = true

    variations {
        value = false
    }

    variations {
        value = true
    }

    defaults {
        off_variation = 0
        on_variation  = 1
    }

    client_side_availability {
        using_environment_id = true
    }
}

resource "launchdarkly_feature_flag_environment" "my-flag-configuration" {
    flag_id        = launchdarkly_feature_flag.my-flag.id
    env_key       = var.env
    on            = true
    off_variation = 0

    targets {
        values    = var.ld-my-flag-orgs
        variation = 1
    }

    fallthrough {
        variation = 0
    }
}
```
When deploying this configuration, the feature flag's behavior is set to false when targeting is off. If you remove the `my-flag-configuration` and redeploy, the behavior will be updated to serve true when targeting is off.
The issue could be related to this [line](https://github.com/launchdarkly/terraform-provider-launchdarkly/blob/main/launchdarkly/resource_launchdarkly_feature_flag_environment.go#L253).
Thank you.

**Comment by sloloris** [comment]
Hi @hamamou,

Thank you for bringing this to our attention!

By design, since you cannot actually delete a feature flag environment configuration, what we do when you delete the resource is patch the environment configuration to match the main default configuration and turn the flag in the given environment off. 

However, as you have correctly identified, it seems we have a bug where we are in fact setting the default off variation value to the last flag variation, which would be the default off variation value if it had not been manually overwritten. We should be setting this to the default off variation. We will address this this week and update you here.

**Comment by sloloris** [comment]
Hi @hamamou!

I'm happy to let you know that we have addressed this issue and it will be included in the next release of our provider. I'm tackling a few more issues this week and will attempt a release either this week or next week, but if this is a blocker for you, let me know and I am happy to do a smaller release beforehand.

Hope your week is going well :)

**Comment by hamamou** [comment]
Fantastic news, That's perfectly fine, we can wait until next week for the release. Thank you so much.

**Comment by sloloris** [comment]
Hi @hamamou ! Just a quick update to say that this has been released in v2.18.0. Please do let us know if you have any other issues and have a great weekend!
